pkg/controller/daemon/daemon_controller.go

```diff
@@ -1469,7 +1469,7 @@ func (dsc *DaemonSetsController) syncNodeUpdate(ctx context.Context, nodeName st
 		daemonPods := podsByDS[dsKey]
 		scheduled := len(daemonPods) > 0
 
-		if (shouldRun && !scheduled) || (!shouldContinueRunning && scheduled) {
+		if shouldRun || scheduled {
```

Comment:
This seems unnecessarily strong and could result in superfluous syncs. Would the following work for you?

```suggestion
		if (shouldRun && !scheduled) || (!shouldContinueRunning && scheduled) || (scheduled && ds.Status.NumberMisscheduled > 0 ) {
```

It would be good to add some explanation above this change and test it with at least a unit test (`TestUpdateNode`).

Reply:
@atiratree Thanks for the review. Code updated and unit test added.